.github/workflows/benchmark_trigger.yml

```
@@ -75,15 +83,42 @@ jobs:
             > "${RUN_JSON}"
           echo "workflow-url=$(jq '.url' ${RUN_JSON})" >> "${GITHUB_OUTPUT}"
           echo "run-status=$(jq --raw-output '.status' ${RUN_JSON})" >> "${GITHUB_OUTPUT}"
+          echo "run-started-at=$(jq --raw-output '.run_started_at' ${RUN_JSON})" >> "${GITHUB_OUTPUT}"
+
+      - name: "Checking if the workflow has been rerun"
+        id: check
+        if: fromJSON(steps.find-workflow.outputs.workflow-url) != null
+        env:
+          RUN_STARTED_AT: ${{ steps.find-workflow.outputs.run-started-at }}
+          TRIGGERED_AT: ${{ needs.precondition.outputs.triggered-at }}
+        run: |
+          # If the latest workflow run started after the trigger event, it means
+          # the workflow has been rerun and picked up the new labels. Skip rerun
+          # in this case.
+          RUN_STARTED_AT_EPOCH="$(date --date="${RUN_STARTED_AT}" +%s)"
+          if (( "${RUN_STARTED_AT_EPOCH}" < "${TRIGGERED_AT}" )); then
```

You actually don't need to do explicit expansion inside arithmetic comparisons

```suggestion
          if (( RUN_STARTED_AT_EPOCH < TRIGGERED_AT )); then
```